### PR TITLE
Add jitter option

### PR DESCRIPTION
**Pull request by jeffreymeng**
Adds an opt-in jitter option, which selects a time uniformly randomly from the interval [0.5, 1.0)*delay each time it pauses. The randomness is applied only to the sleep, and doesn't affect how much future delays increase by.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ edition = "2021"
 [dependencies]
 tokio = { version = "1.24.2", features = ["time"] }
 tracing = "0.1.32"
+rand = "0.8.5"
```
Comment (ekzhang) [inline Cargo.toml]: I would use [`fastrand`](https://docs.rs/fastrand/latest/fastrand/) if you don't need any advanced features! :D

File: src/lib.rs [hunk 2]
```diff
@@ -57,6 +61,21 @@ impl Retry {
...
+    }
+
+    fn apply_jitter(&self, delay: Duration) -> Duration {
+        if self.enable_jitter {
+            // [0.5, 1.0)
+            delay.mul_f64(0.5 + rand::random::<f64>() / 2.0)
+        } else {
+            delay
+        }
+    }
```
Comment (ekzhang) [inline src/lib.rs]: Nice, it seems consistent with industry standards https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/EqualJitterBackoffStrategy.html

**Review by ekzhang — APPROVED** [review]

**Comment by jeffreymeng** [comment]
@ekzhang do you mind publishing the new version?

**Comment by ekzhang** [comment]
Will do!

**Comment by ekzhang** [comment]
Should be published now. https://crates.io/crates/named-retry

**Pull request merged**